### ISSUE TITLE
'no such image' error when trying to schedule pods on non master nodes

### ISSUE THREAD

**Issue by rawlingsj** [openshift/origin#3928]
Origin 1.0.3, I've got a multi-node setup with one master (which is also a schedulable node) and two further nodes.

The ansible origin installer (v3.0.0-7) ran successfully and the logs of both nodes successfully connect to the master API server when starting.  

If I run any pods they only work if they are pinned to the master node using a node selector.  In the logs of the nodes I can see an error message saying 'no image found', I'm not sure which image but they are definitely all available for the pods that I'm trying to run.  As soon as the pod is scheduled on the master node then it runs successfully.  This happened not only for a custom image as seen in the logs below but also for the registry and router.

``````
Desired: []*api.Pod{(*api.Pod)(0xc208b54780), (*api.Pod)(0xc208b54b40)}
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267088   16112 volumes.go:96] Used volume plugin "kubernetes.io/secret" for (volume.Spec){Name:(string)default-token-9w61f VolumeSource:(api.VolumeSource){HostPath:(*api.HostPathVolumeSource)<nil> EmptyDir:(*api.EmptyDirVolumeSource)<nil> GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)<nil> AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSou
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267088   16112 volumes.go:96] Used volume plugin "kubernetes.io/secret" for (volume.Spec){Name:(string)default-token-9w61f VolumeSource:(api.VolumeSource){HostPath:(*api.HostPathVolumeSource)<nil> EmptyDir:(*api.EmptyDirVolumeSource)<nil> GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)<nil> AWSElasticBlockStore:(*api.AWSElasticBlockStoreVolumeSou
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267145   16112 kubelet.go:2054] Generating status for "hubot-letschat-v9rvr_default"
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267151   16112 kubelet.go:2054] Generating status for "letschat-kfsid_default"
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267386   16112 volumes.go:157] Making a volume.Cleaner for /var/lib/openshift/openshift.local.volumes/pods/033b0d3f-3531-11e5-a23a-fa163e55a403/volumes/kubernetes.io~secret
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267401   16112 volumes.go:194] Used volume plugin "kubernetes.io/secret" for 033b0d3f-3531-11e5-a23a-fa163e55a403/kubernetes.io~secret
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267438   16112 volumes.go:157] Making a volume.Cleaner for /var/lib/openshift/openshift.local.volumes/pods/03435abd-3531-11e5-a23a-fa163e55a403/volumes/kubernetes.io~secret
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.267445   16112 volumes.go:194] Used volume plugin "kubernetes.io/secret" for 03435abd-3531-11e5-a23a-fa163e55a403/kubernetes.io~secret
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.274720   16112 manager.go:1288] Syncing Pod &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:hubot-letschat-v9rvr GenerateName:hubot-letschat- Namespace:default SelfLink:/api/v1/namespaces/default/pods/hubot-letschat-v9rvr UID:03435abd-3531-11e5-a23a-fa163e55a403 ResourceVersion:3026 CreationTimestamp:2015-07-28 10:00:32 -0400 EDT DeletionTimestamp:<nil>
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: ecrets/kubernetes.io/serviceaccount}] LivenessProbe:<nil> ReadinessProbe:<nil> Lifecycle:<nil> TerminationMessagePath:/dev/termination-log ImagePullPolicy:IfNotPresent SecurityContext:0xc208ec8280}] RestartPolicy:Always TerminationGracePeriodSeconds:<nil> ActiveDeadlineSeconds:<nil> DNSPolicy:ClusterFirst NodeSelector:map[] ServiceAccount:default Host:cd-node1.novaloc
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.274837   16112 manager.go:1308] Need to restart pod infra container for "hubot-letschat-v9rvr_default" because it is not found
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.274916   16112 manager.go:1327] Container {Name:hubot-letschat Image:docker.io/fabric8/hubot-letschat:v1.0.0 Command:[] Args:[] WorkingDir: Ports:[{Name:http HostPort:0 ContainerPort:8080 Protocol:TCP HostIP:}] Env:[{Name:LETSCHAT_HUBOT_USERNAME Value:fabric8 ValueFrom:<nil>} {Name:HUBOT_LCB_ROOMS Value:fabric8_default ValueFrom:<nil>} {Name:HUBOT_GRAFA
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.274960   16112 manager.go:1434] Got container changes for pod "hubot-letschat-v9rvr_default": {StartInfraContainer:true InfraContainerId: ContainersToStart:map[0:{}] ContainersToKeep:map[]}
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.274975   16112 manager.go:1443] Killing Infra Container for "hubot-letschat-v9rvr_default", will start new one
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.274982   16112 manager.go:1468] Creating pod infra container for "hubot-letschat-v9rvr_default"
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.275890   16112 docker.go:136] Pulling image openshift/origin-pod:v3.0.0.1 without credentials
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.281745   16112 manager.go:1288] Syncing Pod &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:letschat-kfsid GenerateName:letschat- Namespace:default SelfLink:/api/v1/namespaces/default/pods/letschat-kfsid UID:033b0d3f-3531-11e5-a23a-fa163e55a403 ResourceVersion:3016 CreationTimestamp:2015-07-28 10:00:32 -0400 EDT DeletionTimestamp:<nil> Labels:map[name:l
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: ueFrom:<nil>} {Name:LETSCHAT_HUBOT_DISPLAY_NAME Value:fabric8 ValueFrom:<nil>} {Name:LETSCHAT_DEFAULT_ROOMS Value:fabric8_default ValueFrom:<nil>} {Name:LCB_NOROBOTS Value:false ValueFrom:<nil>} {Name:LCB_FILES_ENABLE Value:true ValueFrom:<nil>} {Name:LCB_DATABASE_URI Value:mongodb://127.0.0.1:27017/letschat ValueFrom:<nil>}] Resources:{Limits:map[] Requests:map[]} Vo
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.281850   16112 manager.go:1308] Need to restart pod infra container for "letschat-kfsid_default" because it is not found
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.281891   16112 manager.go:1327] Container {Name:letschat-mongodb Image:mongo Command:[] Args:[] WorkingDir: Ports:[{Name: HostPort:0 ContainerPort:27017 Protocol:TCP HostIP:}] Env:[] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[{Name:default-token-9w61f ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount}] LivenessProbe:<nil>
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.281960   16112 manager.go:1327] Container {Name:letschat Image:fabric8/lets-chat Command:[] Args:[] WorkingDir: Ports:[{Name: HostPort:0 ContainerPort:5000 Protocol:TCP HostIP:}] Env:[{Name:LETSCHAT_CREATE_HUBOT_USER Value:true ValueFrom:<nil>} {Name:LETSCHAT_HUBOT_TOKEN Value:${LETSCHAT_HUBOT_TOKEN} ValueFrom:<nil>} {Name:LETSCHAT_HUBOT_PASSWORD Value:
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.282004   16112 manager.go:1434] Got container changes for pod "letschat-kfsid_default": {StartInfraContainer:true InfraContainerId: ContainersToStart:map[0:{} 1:{}] ContainersToKeep:map[]}
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.282016   16112 manager.go:1443] Killing Infra Container for "letschat-kfsid_default", will start new one
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.282022   16112 manager.go:1468] Creating pod infra container for "letschat-kfsid_default"
Jul 28 10:19:43 cd-node1.novalocal openshift-node[16112]: I0728 10:19:43.283083   16112 docker.go:136] Pulling image openshift/origin-pod:v3.0.0.1 without credentials
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: I0728 10:19:46.619176   16112 server.go:569] Event(api.ObjectReference{Kind:"Pod", Namespace:"default", Name:"letschat-kfsid", UID:"033b0d3f-3531-11e5-a23a-fa163e55a403", APIVersion:"v1", ResourceVersion:"3016", FieldPath:"implicitly required container POD"}): reason: 'pulled' Successfully pulled image "openshift/origin-pod:v3.0.0.1"
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: I0728 10:19:46.619695   16112 manager.go:602] Container default/letschat-kfsid/POD: setting entrypoint "[]" and command "[]"
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: E0728 10:19:46.622363   16112 manager.go:1476] Failed to create pod infra container: no such image; Skipping pod "letschat-kfsid_default"
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: I0728 10:19:46.622384   16112 kubelet.go:2054] Generating status for "letschat-kfsid_default"
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: I0728 10:19:46.622426   16112 server.go:569] Event(api.ObjectReference{Kind:"Pod", Namespace:"default", Name:"letschat-kfsid", UID:"033b0d3f-3531-11e5-a23a-fa163e55a403", APIVersion:"v1", ResourceVersion:"3016", FieldPath:"implicitly required container POD"}): reason: 'failed' Failed to create docker container with error: no such image
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: I0728 10:19:46.633334   16112 status_manager.go:103] Ignoring same pod status for letschat-kfsid_default - old: {Pending [{Ready False}]  42.42.42.91  2015-07-28 10:00:40.928711068 -0400 EDT [{letschat-mongodb {%!s(*api.ContainerStateWaiting=&{Image: mongo is ready, container is creating}) %!s(*api.ContainerStateRunning=<nil>) %!s(*api.ContainerStateTerminated=<nil>)}
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: E0728 10:19:46.633439   16112 pod_workers.go:108] Error syncing pod 033b0d3f-3531-11e5-a23a-fa163e55a403, skipping: no such image
Jul 28 10:19:46 cd-node1.novalocal openshift-node[16112]: I0728 10:19:46.633570   16112 server.go:569] Event(api.ObjectReference{Kind:"Pod", Namespace:"default", Name:"letschat-kfsid", UID:"033b0d3f-3531-11e5-a23a-fa163e55a403", APIVersion:"v1", ResourceVersion:"3016", FieldPath:""}): reason: 'failedSync' Error syncing pod, skipping: no such image
lines 16990-17058/17058 (END)```
``````

**Comment by ironcladlou** [comment]
@rawlingsj 

How were the pods created? Directly, or via a deploymentConfig? Could you please post your pod and/or deploymentConfig YAML (`oc get pod -o yaml` and `oc get dc -o yaml`)?

**Comment by rawlingsj** [comment]
Actually both, the pods I was using were created manually but it also happened with the router and registry so they use a deployment config right?

The environment is long gone now unfortunately so don't have the yaml or json but as it was the default 1.0.3 registry and router then maybe that helps?

**Comment by ironcladlou** [comment]
@rawlingsj 

Can you give me more information about the environment to reproduce the problem in? Which ansible setup did you use (e.g. AWS, local, something DIY)? The way I'm interpreting your description so far is "using the standard Origin installation, the router and registry are broken out of the box". Is that basically accurate? You mentioned that the router and registry failed to deploy.

**Comment by rawlingsj** [comment]
@ironcladlou 

Yes sure, it was an OpenStack environment.

I'm not sure what the standard Origin installation is for multi-node setup as the docs seem to only suggest using a docker image for Origin https://github.com/openshift/origin#getting-started, there is every chance I've missed the right doc.

From memory I used the tagged version (3.0.0-3) from https://github.com/openshift/openshift-ansible/tree/v3.0.0-3 and ran this on a four instance (one master, 3 nodes) OpenStack environment.

I modified `/etc/ansible/hosts` and changed 

```
deployment_type=origin
```

I used the steps from this doc but as I said this may well not have been the right one https://github.com/openshift/openshift-ansible/blob/v3.0.0-3/README_origin.md
